### PR TITLE
[KOGITO-7922] - Update Kogito Operator fo Go 1.17

### PR DESCRIPTION
**Pull request by spolti**
Signed-off-by: spolti <fspolti@redhat.com>

### CODE REVIEW

**Review by desmax74 — APPROVED** [review]

File: kogito-operator-jenkins-node/modules/golang-installer/1.17.x/module.yaml [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+version: "1.17"
+description: "Module used to install golang 1.17"
+envs:
+  - name: GOPATH
+    value: /home/jenkins/go
+  - name: PATH
+    value: /home/jenkins/go/bin:$PATH
+artifacts: 
+  - name: go1.17.linux-amd64.tar.gz
+    url: https://dl.google.com/go/go1.16.linux-amd64.tar.gz
```
Comment (radtriste) [inline kogito-operator-jenkins-node/modules/golang-installer/1.17.x/module.yaml]: please update link to go1.17

**Comment by radtriste** [comment]
can you also update the image version to 1.28 ?

**Comment by radtriste** [comment]
created image `quay.io/tradisso/kogito-operator-jenkins-node:1.28.0`

EDIT: Gonna create a new label with this image and update the Jenkinsfile on this PR

File: kogito-operator-jenkins-node/modules/golang-installer/1.17.x/module.yaml (hunk 1 above)
Comment (spolti) [reply kogito-operator-jenkins-node/modules/golang-installer/1.17.x/module.yaml]: oops, thanks.

File: kogito-operator-jenkins-node/modules/jenkins-user/create.sh [hunk 2]
```diff
@@ -2,3 +2,8 @@
 
 set -e
 groupadd -r jenkins -g 1001 &&  useradd -u 1001 -r -g jenkins  -m -d /home/jenkins -c "jenkins user" jenkins
+usermod --add-subuids 10000-75535 jenkins
+usermod --add-subgids 10000-75535 jenkins
+
+cat /etc/subuid
```
Comment (spolti) [inline kogito-operator-jenkins-node/modules/jenkins-user/create.sh]:
```suggestion
cat /etc/subuid

chmod 4755 /usr/bin/newgidmap
chmod 4755 /usr/bin/newuidmap
```

File: kogito-operator-jenkins-node/modules/jenkins-user/create.sh [hunk 3]
```diff
@@ -2,3 +2,11 @@
...
+
+chmod 4755 /usr/bin/newgidmap
+chmod 4755 /usr/bin/newuidmap
+cat /etc/subgid
```
Comment (spolti) [inline kogito-operator-jenkins-node/modules/jenkins-user/create.sh]:
```suggestion
cat /etc/subgid

rpm -qV shadow-utils
getcap /usr/bin/newuidmap /usr/bin/newgidmap

```

File: kogito-operator-jenkins-node/modules/jenkins-user/create.sh [hunk 4]
```diff
@@ -2,3 +2,15 @@
...
+
+rpm -qV shadow-utils
+getcap /usr/bin/newuidmap /usr/bin/newgidmap
```
Comment (spolti) [inline kogito-operator-jenkins-node/modules/jenkins-user/create.sh]:
```suggestion
#getcap /usr/bin/newuidmap /usr/bin/newgidmap
```

File: kogito-operator-jenkins-node/modules/jenkins-user/create.sh [hunk 5]
```diff
@@ -2,3 +2,18 @@
 
 set -e
 groupadd -r jenkins -g 1001 &&  useradd -u 1001 -r -g jenkins  -m -d /home/jenkins -c "jenkins user" jenkins
+usermod --add-subuids 10000-75535 jenkins
+usermod --add-subgids 10000-75535 jenkins
```
Comment (radtriste) [inline kogito-operator-jenkins-node/modules/jenkins-user/create.sh:7]:
```suggestion
usermod --add-subuids 100000-75535 jenkins
usermod --add-subgids 100000-75535 jenkins
```

File: kogito-operator-jenkins-node/modules/jenkins-user/create.sh [hunk 6]
```diff
@@ -2,3 +2,18 @@
 
 set -e
 groupadd -r jenkins -g 1001 &&  useradd -u 1001 -r -g jenkins  -m -d /home/jenkins -c "jenkins user" jenkins
+usermod --add-subuids 100000-75535 jenkins
+usermod --add-subgids 100000-75535 jenkins
```
Comment (radtriste) [inline kogito-operator-jenkins-node/modules/jenkins-user/create.sh]:
```suggestion
usermod --add-subuids 100000-165535 jenkins
usermod --add-subgids 100000-165535 jenkins
```

File: kogito-operator-jenkins-node/modules/jenkins-user/create.sh [hunk 7]
```diff
@@ -2,3 +2,18 @@
 
 set -e
 groupadd -r jenkins -g 1001 &&  useradd -u 1001 -r -g jenkins  -m -d /home/jenkins -c "jenkins user" jenkins
+usermod --add-subuids 100000-75535 jenkins
```
Comment (spolti) [inline kogito-operator-jenkins-node/modules/jenkins-user/create.sh]:
```suggestion
usermod --add-subuids 100000-65536 jenkins
```

File: kogito-operator-jenkins-node/modules/jenkins-user/create.sh (hunk 6 above)
Comment (spolti) [inline kogito-operator-jenkins-node/modules/jenkins-user/create.sh]:
```suggestion
usermod --add-subgids 100000-65536 jenkins
```

File: kogito-operator-jenkins-node/modules/jenkins-user/create.sh (hunk 7 above)
Comment (spolti) [inline kogito-operator-jenkins-node/modules/jenkins-user/create.sh]:
```suggestion
id

usermod --add-subuids 100000-75535 jenkins
```

File: kogito-operator-jenkins-node/modules/jenkins-user/create.sh [hunk 8]
```diff
@@ -2,3 +2,18 @@
...
+
+cat /etc/subuid
+
+chmod 4755 /usr/bin/newgidmap
+chmod 4755 /usr/bin/newuidmap
+cat /etc/subgid
+
+#rpm -qV shadow-utils
+#getcap /usr/bin/newuidmap /usr/bin/newgidmap
+
```
Comment (spolti) [inline kogito-operator-jenkins-node/modules/jenkins-user/create.sh:19]: id

File: kogito-operator-jenkins-node/modules/jenkins-user/create.sh [hunk 9]
```diff
@@ -2,3 +2,18 @@
...
```
Comment (spolti) [inline kogito-operator-jenkins-node/modules/jenkins-user/create.sh:5]:
```suggestion
id
podman unshare cat /proc/self/uid_map

```

File: kogito-operator-jenkins-node/modules/podman/config/containers.conf [hunk 10]
```diff
@@ -5,8 +5,7 @@ userns="host"
 ipcns="host"
 utsns="host"
 cgroupns="host"
-cgroups="disabled"
 log_driver = "k8s-file"
 [engine]
 cgroup_manager = "cgroupfs"
-events_logger="file"
\ No newline at end of file
+events_logger="file""
```
Comment (spolti) [inline kogito-operator-jenkins-node/modules/podman/config/containers.conf]: extra "

**Review by radtriste — APPROVED** [review]

**Pull request merged**